packages/x-adapter/jest.config.js

```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  moduleFileExtensions: ['ts', 'js'],
+  transform: {
+    '^.+\\.ts?$': 'ts-jest'
```

Comment:
Reading this article: https://miyauchi.dev/posts/speeding-up-jest/

We test to use different transformers to improve the perfomance:

ts-jest:

![image](https://user-images.githubusercontent.com/6597815/158571533-e60bc626-fea2-491a-aed4-b64a319d1941.png)

@swc/jest:

![image](https://user-images.githubusercontent.com/6597815/158571605-96ebfe0a-00de-431f-a5b8-bbc8642bb27d.png)

esbuild-jest:

![image](https://user-images.githubusercontent.com/6597815/158571656-c92aba67-c5cf-4494-b168-1f2bcfba038e.png)

I dont know if we should give a try to improve the jest performance.

Reply:
It is really great to see that you are exploring alternative tools, and properly testing them, but TBH, I don't think we have any performance issue with jest. 974 tests and it takes 20s to execute in my laptop, I've seen yesterday 50s worst scenario in yours, and it went down up to 35s with tweaks. There is no much room of improvement there, just a couple of seconds. 

Introducing another compiler as esbuild or swc here would make us lose type checking, which is far more valuable for me than saving 20s of the build time. 

Also, we need to be careful with these compilers. This are alternative versions to the TS one, which means that they will probably output different code, or they might not even support al language features.

Every tool has its own pros and cons, and what I'm trying to say here is that if we introduce a new tool, we must make sure that we are maximising its pros, and minimising its cons. In this particular case I don't think it is worth to lose type checking and to risk losing future TypeScript features just to save a few seconds in test time. Indeed, we can currently speed up our tests by disabling type checking, which is really expensive by adding  the config below to `jest.config.js`, and I still wouldn't do it in this case.

```js
  globals: {
    'ts-jest': {
      isolatedModules: true
    }
  }
```

Reply:
<img src="https://media2.giphy.com/media/BEob5qwFkSJ7G/giphy.gif"/>